app/assets/javascripts/discourse/app/static/wizard/components/wizard-step.gjs

```diff
@@ -0,0 +1,302 @@
+              </button>
+            {{/if}}
+          </div>
+
+          <div class="wizard-container__buttons-right">
+            {{#if this.showFinishButton}}
+              <button
+                {{on "click" this.finish}}
+                disabled={{this.saving}}
+                type="button"
+                class="wizard-container__button finish"
+              >
+                {{i18n "wizard.finish"}}
+              </button>
+            {{else if this.showConfigureMore}}
+              <button
+                {{on "click" this.nextStep}}
+                disabled={{this.saving}}
+                type="button"
+                class="wizard-container__button configure-more"
+              >
+                {{i18n "wizard.configure_more"}}
+              </button>
+            {{/if}}
+
+            {{#if this.showJumpInButton}}
+              <button
+                {{on "click" this.jumpIn}}
+                disabled={{this.saving}}
+                type="button"
+                class="wizard-container__button primary jump-in"
+              >
+                {{i18n "wizard.jump_in"}}
+              </button>
+            {{else}}
+              <button
+                {{on "click" this.nextStep}}
+                disabled={{this.saving}}
+                type="button"
+                class="wizard-container__button primary next"
+              >
+                {{i18n "wizard.next"}}
+              </button>
+            {{/if}}
+
+          </div>
+
+        </div>
+      </div>
+    </div>
+  </template>
+
+  @tracked saving = false;
+
+  get wizard() {
+    return this.args.wizard;
+  }
+
+  get step() {
+    return this.args.step;
+  }
+
+  get id() {
+    return this.step.id;
+  }
+
+  /**
+    Step        Back Button?     Primary Action      Secondary Action
+    ------------------------------------------------------------------
+    First            No               Next                  N/A
+    ------------------------------------------------------------------
+    ...             Yes               Next                  N/A
+    ------------------------------------------------------------------
+    Ready           Yes              Jump In          Configure More
+    ------------------------------------------------------------------
+    ...             Yes               Next              Exit Setup
+    ------------------------------------------------------------------
+    Last            Yes              Jump In                N/A
+    ------------------------------------------------------------------
+
+    Back Button: without saving, go back to the last page
+    Next Button: save, and if successful, go to the next page
+    Configure More: re-skinned next button
+    Exit Setup: without saving, go to the home page ("finish")
+    Jump In: on the "ready" page, it exits the setup ("finish"), on the
+    last page, it saves, and if successful, go to the home page
+
+   */
```

Comment:
This is actually a considerable improvement over the status quo, in that the only step that we sort of hardcode is "ready". It will work just fine if that step isn't present, but if it does, it splits the wizard flow into the "required" and "optional" steps and displays the buttons correctly, without hard coding in "styling", "branding", etc. And the final step is now only "special" by way of being last, rather than hard coding "corporate".